ghost/core/core/server/data/migrations/versions/5.89/2024-07-30-19-51-06-backfill-offer-redemptions.js

```diff
@@ -0,0 +1,62 @@
+// For information on writing migrations, see https://www.notion.so/ghost/Database-migrations-eb5b78c435d741d2b34a582d57c24253
+
+const logging = require('@tryghost/logging');
+const {default: ObjectID} = require('bson-objectid');
+
+// For DML - data changes
+const {createTransactionalMigration} = require('../../utils');
+
+module.exports = createTransactionalMigration(
+    async function up(knex) {
+        // Backfill missing offer redemptions
+        try {
+            // Select all subscriptions that have an `offer_id` but don't have a matching row in the `offer_redemptions` table
+            logging.info('Selecting subscriptions with missing offer redemptions');
+            const result = await knex.raw(`
+                SELECT
+                    mscs.id AS subscription_id,
+                    mscs.offer_id,
+                    mscs.start_date AS created_at,
+                    m.id AS member_id
+                FROM
+                    members_stripe_customers_subscriptions mscs
+                        JOIN 
+                    members_stripe_customers msc ON mscs.customer_id = msc.customer_id
+                        JOIN
+                    members m ON msc.member_id = m.id
+                        LEFT JOIN
```

Comment:
Even though I think the query optimizer handles this, moving this join first makes sense as it is the primary and only join to filter.

Reply:
Ah, nice one — TIL! 🙏🏼